lib/_http_incoming.js

```diff
@@ -77,6 +78,7 @@ function IncomingMessage(socket) {
   this.rawTrailers = [];
 
   this.aborted = false;
+  this[kAbortController] = null;
```

Comment:
should we provide some access to the `AbortController`?

Reply:
If we're only looking at the use case presented here, I don't think so, but there's a larger consideration here that makes me question this PR. The approach here uses the abort on the request as the trigger, but that's not always reliable and ignores the other reasons why a request flow may be aborted. There are other scenarios where a request flow would be aborted where aborting the actual streams may not be desirable. In those cases, it would be better for the AbortController to exist independent of the Request object, and for the Request to optionally register itself as an `abort` event listener (rather than the other way around seen in this PR).

tl;dr ... I'm not sure if this PR is the right approach.

Reply:
IMO, putting aside the correctness of my code, I was wondering if the signal should be limited to just aborted requests - maybe the signal should notify on completion? I can see how that would be useful as well.

Reply:
I think that if the AbortCobtroller will be exposed then it should be a method and not a getter.

That said I think not exposing it makes sense too.

I also am not sure regarding whether it should signal completion or specifically "aborted"

Reply:
Well, if the `signal` is just limited to aborted requests, the request object *already* exposes an `aborted` event that user land can tap in to (just like this PR does). The `AbortSignal` just becomes redundant.

Reply:
> The AbortSignal just becomes redundant

I think the feature request was since other APIs take AbortSignals this would let you pass the signal to those APIs directly instead of creating an AbortController, listening to `"aborted"` etc. If everything takes an AbortSignal you can just pass signals and not worry about events all the way which makes it easier to write safer and "obviously correct" code.

That said I have no idea if it should listen to completion or just "aborted" 😅

Reply:
Other platforms also have similar feature. For example [.NET](https://docs.microsoft.com/en-us/aspnet/core/mvc/models/model-binding?view=aspnetcore-3.1#cancellationtoken)) can either [DI a CancellationToken](https://andrewlock.net/using-cancellationtokens-in-asp-net-core-mvc-controllers/#using-cancellationtokens-in-your-mvc-actions) or be used directly from [HttpContext.RequestAborted](https://docs.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.http.httpcontext.requestaborted?view=aspnetcore-5.0)